notebooks/214-vehicle-detection-and-recognition/214-vehicle-detection-and-recognition.ipynb

```diff
@@ -0,0 +1,545 @@
+    "plt_show(cv2.cvtColor(resized_image_re, cv2.COLOR_BGR2RGB))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8a70376f",
+   "metadata": {},
+   "source": [
+    "##### Recognition progressing\n",
+    "\n",
+    "We will get the result contains colors(White, gray, yellow, red, green, blue, black) and types(Car, bus, truck, van), then we need to get the probability of each attribute, finally we need to choose the max probability as the result."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f2159ec4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def vehicle_recognition(exec_net_re, input_size, raw_image):\n",
+    "    \"\"\"\n",
+    "    Vehicle attributes recognition, input a single vehicle, return it's attribute\n",
+    "    :param: exec_net_re: recognition net \n",
+    "    :param: input_size: recognition input size\n",
+    "    :param: raw_image: single vehicle image\n",
+    "    :returns: attr_color: predicted color\n",
+    "              attr_type: predicted type\n",
+    "    \"\"\"\n",
+    "    # vehicle's attribute\n",
+    "    colors = ['White', 'Gray', 'Yellow', 'Red', 'Green', 'Blue', 'Black']\n",
+    "    types = ['Car', 'Bus', 'Truck', 'Van']\n",
+    "    \n",
+    "    # resize image to input size\n",
+    "    resized_image_re = cv2.resize(raw_image, input_size)\n",
+    "    input_image_re = np.expand_dims(resized_image_re.transpose(2, 0, 1), 0)\n",
+    "    plt_show(cv2.cvtColor(resized_image_re, cv2.COLOR_BGR2RGB))\n",
+    "    \n",
+    "    # inference the recognition net\n",
+    "    result = exec_net_re.infer(inputs={input_key_re[0]: input_image_re})\n",
+    "    \n",
+    "    # predict result\n",
+    "    predict_colors = result['color']\n",
+    "    # delete the dim of 2, 3\n",
+    "    predict_colors = np.squeeze(predict_colors, (2, 3))\n",
+    "    predict_types = result['type']\n",
+    "    predict_types = np.squeeze(predict_types, (2, 3))\n",
+    "\n",
+    "    attr_color, attr_type = (colors[np.argmax(softmax(predict_colors))],\n",
+    "                             types[np.argmax(softmax(predict_types))])\n",
+    "    return attr_color, attr_type"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "79505e68",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f\"Attributes:{vehicle_recognition(exec_net_re, (72, 72), test_car)}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "54075f4e",
+   "metadata": {},
+   "source": [
+    "### Conbine two models\n",
+    "\n",
+    "Congratulations!Now we can use detection model to crop single vehicle and recognize the vehicle's attribute."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "59402e0d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def convert_result_to_image(exec_net_re, bgr_image, resized_image, boxes, threshold=0.6):\n",
```

Comment:
This function is almost the same as the previous one. Please avoid code repetition